### ISSUE TITLE
TsCoverageSensor not doing anything

### ISSUE THREAD

**Issue by aelnelson** [Pablissimo/SonarTsPlugin#76]
I'm using version 0.4-SNAPSHOT to get coverage of my typescript files with sonar-scanner-2.7, however the TsCoverageSensor isn't doing anything. I've run the sonar-scanner with the debug (-X) flag and it doesn't output anything in the TsCoverageSensor. It just has this 

    INFO: Sensor TsCoverageSensor
    INFO: Sensor TsCoverageSensor (done) | time=5ms

This is my sonar-project.properties file (with fields sanitized)

```# required metadata
sonar.projectKey=name
sonar.projectName=name
sonar.projectVersion=1.0
sonar.projectDescription=description

# path to source directories
sonar.sources=folder1/src/main, folder2/src/main, folder3/src/main
sonar.tests=folder1/src/test, folder2/src/test, folder3/src/test

# language of the project
sonar.language=ts

# encoding of the source code
sonar.sourceEncoding=UTF-8

# coverage
sonar.ts.lcov.reportPath=coverage/lcov.info
sonar.ts.jstest.reportsPath=coverage
sonar.genericcoverage.unitTestReportPaths=coverage/TEST-xunit.xml```

The coverage folder/lcov.inf file definitely exist, but it doesn't seem to even be trying to run it.

**Comment by aelnelson** [comment]
For reference, the coverage information is generated by mocha and instanbul on the javascript, and the javascript is then mapped back to the typescript using remap-instanbul.

**Comment by nixel2007** [comment]
I have no problem with 0.96-PRE version on SQ 6.1 and typescript project.
![image](https://cloud.githubusercontent.com/assets/1132840/21179741/7b3536fe-c206-11e6-9816-f156e59866c3.png)

```
sonar.ts.lcov.reportpath=coverage/lcov.info
```

**Comment by aelnelson** [comment]
@nixel2007 
What shows up for you in the TsCoverageSensor section when you run it? Is it also blank?

 I've been thinking it might be a version issue, but I can't upgrade to too recent a version (I believe 0.9-PRE is the latest I can do) as I don't have permission to upgrade the SonarQube version.

**Comment by nixel2007** [comment]
@aelnelson I have no TsCoverageSensor in logs. But there was something like "LCOV and LOC sensor" and it says it found lcof.info and start analyzis. 

Yes, I use mocha & istanbul & and remap-istanbul
https://github.com/xDrivenDevelopment/vsc-language-1c-bsl/blob/develop/test/index.ts

**Comment by nixel2007** [comment]
Also 0.96 should work on 5.6 SQ, cause it uses the 5.6 sonar-plugin-api

**Comment by Pablissimo** [comment]
Did you restart SonarQube after upgrading the plugin?

**Comment by nixel2007** [comment]
Sure (if this question is for me)

**Comment by Pablissimo** [comment]
Nope, for @aelnelson :)

**Comment by aelnelson** [comment]
@Pablissimo 
I haven't actually been able to update the version, I also need to get permission to update it. My boss wants me to try to get this version to work first before updating. Or would your advice be to just try to update it to 0.9 and see if that works?

In regards to my current version, should there be any output in debug mode? It's hard to figure out what's wrong when there's no logs.

**Comment by Pablissimo** [comment]
I would try it locally, certainly - you can download Sonarqube such that it just runs from a folder against an embedded database, then download the scanner and run it against your local machine. At least that way you don't end up going to your boss arguing for an upgrade that doesn't fix your issue!

0.4 was pretty early doors, and while I'd expect it to work fine it doesn't have quite so detailed logging. Can't guarantee it'll work but it will be your best bet as a starting point if you can run the latest version against a local install.

**Comment by aelnelson** [comment]
After downloading quite a few copies of SQ and your plugin I discovered what was wrong. For the `sonar.ts.lcov.reportpath` property in the sonar-project.properties file we had used camel case for reportpath. This is why it wasn't reporting anything for the TsCoverageSensor, as we hadn't actually set the correct property.
That's completely our fault, although it might be a good idea to use camel case like all of the other sonar project properties to avoid confusion in the future.
